lib/stm32/l4/rcc.c

```diff
@@ -37,12 +37,65 @@
 
 /**@{*/
 #include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/flash.h>
+#include <libopencm3/stm32/pwr.h>
 
 /* Set the default clock frequencies after reset. */
 uint32_t rcc_ahb_frequency = 4000000;
 uint32_t rcc_apb1_frequency = 4000000;
 uint32_t rcc_apb2_frequency = 4000000;
 
+const struct rcc_clock_scale rcc_clock_config[] = {
+	{ /* 80MHz PLL from HSI */
+		.pll_source = RCC_PLLCFGR_PLLSRC_HSI16,
+		.pllm = 1,
+		.plln = 10,
+		.pllp = RCC_PLLCFGR_PLLP_DIV7,
+		.pllq = RCC_PLLCFGR_PLLQ_DIV4,
+		.pllr = RCC_PLLCFGR_PLLR_DIV2,
+		.hpre = RCC_CFGR_HPRE_NODIV,
+		.ppre1 = RCC_CFGR_PPRE1_DIV2,
+		.ppre2 = RCC_CFGR_PPRE2_DIV2,
+		.voltage_scale = PWR_SCALE1,
+		.flash_waitstates = 4,
+		.ahb_frequency = 80000000,
+		.apb1_frequency = 40000000,
+		.apb2_frequency = 40000000
+	},
+	{ /* 80MHz PLL from 24MHz HSE */
+		.pll_source = RCC_PLLCFGR_PLLSRC_HSE,
+		.pllm = 3,
+		.plln = 20,
+		.pllp = RCC_PLLCFGR_PLLP_DIV7,
+		.pllq = RCC_PLLCFGR_PLLQ_DIV4,
+		.pllr = RCC_PLLCFGR_PLLR_DIV2,
+		.hpre = RCC_CFGR_HPRE_NODIV,
+		.ppre1 = RCC_CFGR_PPRE1_DIV2,
+		.ppre2 = RCC_CFGR_PPRE2_DIV2,
+		.voltage_scale = PWR_SCALE1,
+		.flash_waitstates = 4,
+		.ahb_frequency = 80000000,
+		.apb1_frequency = 40000000,
+		.apb2_frequency = 40000000
+	},
+	{ /* 72MHz PLL and 48MHz USB from 24MHz HSE */
```

Comment:
This... works, but better is to just still be 80Mhz, and use pllsaiq for clock48 instead of having to cut down to 72Mhz.